README.md

```diff
@@ -44,45 +44,102 @@ plugins: [
+      // 
+      // Default: true
+      trackPage: true,
+
+      // Boolean indicating if you want this plugin to perform a page() call immediately once the snippet
+      // is loaded.
+      // 
+      // You might want to disable this if you *only* want page() calls to occur upon Client-side routing
+      // updates. See `trackPageOnRouteUpdate` option.
+      // 
+      // This plugin will still attempt to intelligently prevent duplicate page() calls.
+      // 
+      // Default: true
+      trackPageImmediately: true,
 
-      // number (defaults to 50); time to wait after a route update before it should
-      // track the page change, to implement this, make sure your `trackPage` property is set to `true`
-      trackPageDelay: 50,
+      // Boolean indicating whether to ignore `page` calls by this plugin before we call `analytics.load`
+      // and/or the `ready` event has been emitted by `analytics`.
+      // 
+      // Useful in some cases to prevent issues "queuing" `page` events before Segment is fully loaded.
+      trackPageOnlyIfReady: false,
 
-      // Whether or not to include the document.title in the analytics.page() call
+      // Boolean indicating whether to perform any page() calls during Client-side routing updates.
+      // 
+      // You might want to disable `trackPageImmediately` if you *only* want page() calls to occur upon
+      // Client-side routing updates.
+      // 
+      // This plugin will still attempt to intelligently prevent duplicate page() calls.
+      // 
       // Default: true
-      includeTitleInPageCall: true,
+      trackPageOnRouteUpdate: true,
 
-      // If you need to proxy events through a custom endpoint,
-      // add a `host` property (defaults to https://cdn.segment.io)
-      // Segment docs:
-      //   - https://segment.com/docs/connections/sources/custom-domains
-      //   - https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#proxy
-      host: 'https://override-segment-endpoint',
+      // Number indicating what delay (in ms) should be used when calling analytics.page() in response
+      // to a Gatsby `onRouteUpdate` event. This can help to ensure that the segment route tracking is 
+      // in sync with the actual Gatsby route. Otherwise you can end up in a state where the Segment
+      // page tracking reports the previous page on route change.
+      // 
+      // See https://www.gatsbyjs.com/docs/reference/config-files/gatsby-browser/ for more information.
+      // 
+      // Default: 50
+      trackPageOnRouteUpdateDelay: 50,
+
+      // Boolean indicating whether or not to add the document.title as the first argument to
+      // the analytics.page() calls. Segment uses some san defaults, but some users of this plugin
+      // have wanted to do this in the past.
+      // 
+      // E.g `analytics.page(document.title)`
+      // 
+      // Default: false
+      includeTitleInTrackPage: false,
```

Comment:
prob too late to change this but in hindsight, maybe this should've been `trackPageWithTitle` so it's namespaced and easier to find. oh well.

Reply:
This is a `major` release, so we can change names and behaviors that would otherwise have been considered "breaking" without worrying about it.

Reply:
yeah, but even tho it's major i don't think we should use that as an opportunity to change names that are already set. lets just leave it for now... oh well no big deal.